pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentDictionaryCreator.java

```diff
@@ -268,7 +268,7 @@ public int indexOfSV(Object value) {
         throw new UnsupportedOperationException("Unsupported data type : " + _fieldSpec.getDataType());
     }
 
-    Arrays.sort(indexes);
+//    Arrays.sort(indexes);
```

Comment:
Does this impact the list column type as well? 
Can you add some comments on why this should be removed? Also, I'd recommend removing the line than commenting it.